website/docs/components/icon.mdx

```diff
@@ -70,51 +102,90 @@ import { MdSettings } from "react-icons/md"
+    />
+  ),
 })
-```
 
-The major advantage of `createIcon` is that it forwards ref properly.
+// OR Using `pathDefinition` directly
 
-If you use <Icon as={MdSettings}/> and `MdSettings` doesn't use `forwardRef`, the ref doesn't get forwarded automatically. This may report a React error in the console about passing ref to a function component. In short, you can use `<Icon as={MdSettings} />` if you know that `MdSettings` is forwarding the ref - otherwise, use the `createIcon` utility function and then call the icon directly: `<MdSettings />`.
+export const UpDownIcon = createIcon({
+  displayName: "UpDownIcon",
+  viewBox: "0 0 200 200",
+  d: "M 100, 100 m -75, 0 a 75,75 0 1,0 150,0 a 75,75 0 1,0 -150,0",
+})
+```
 
-Before generating your custom icon do the following:
+Before generating custom icons do the following:
 
 - Export the icon's `svg` from Figma, Sketch, etc.
 - Use a tool like [SvgOmg](https://jakearchibald.github.io/svgomg/) to reduce
   the size and minify the markup.
-- Use `generateIcon` to create an icon as shown above.
+- Use `createIcon` to create an icon as shown above.
+
+Start using the icons right away. There is no need to update theme files.
+
+## Fallback Icon
+
+When path is not provided, `Icon` component renders the `fallbackIcon`
 
-Start using your icon right away. There is no need to update your theme files.
+```jsx
+<Icon boxSize="32px" color="white" />
+```
 
 ## Props
 
-Icon component composes `Box` so you can pass all `Box` props.
+### Icon Props
 
 | Name        | Type                  | Default        | Description                                                                         |
 | ----------- | --------------------- | -------------- | ----------------------------------------------------------------------------------- |
-| `size`      | `string`              | `1em`          | The size of the icon.                                                               |
-| `name`      | `string`              |                | The name of the icon.                                                               |
+| `viewBox`   | `string`              | `0 0 24 24`    | The viewBox of the icon.                                                            |
+| `boxSize`   | `string`              | `1em`          | The size of the icon.                                                               |
 | `color`     | `string`              | `currentColor` | The color of the icon.                                                              |
 | `focusable` | `boolean`             | `false`        | Denotes that the icon is not an interative element, and only used for presentation. |
 | `role`      | `presentation`, `img` | `presentation` | The html role of the icon.                                                          |
+| `children`  | `React.ReactNode`     |                | The Path or Group of the icon                                                       |
+
+## createIcon Options
```

Comment:
Since, we are exposing the createIcon utility, thought its best we can add the options at the bottom